### DISCUSSION TITLE
Implement MMagic visualization and write a user guide 实现 MMagic 可视化并编写使用教程

### DISCUSSION THREAD

**Discussion by OpenMMLab-Assistant-004** [open-mmlab/OpenMMLabCamp#302] [MMagic]
# Implement MMagic visualization and write a user guide 
 实现 MMagic 可视化并编写使用教程 

*Please follow these steps:*
1. Study the MMEngine visualization tutorial available at [https://github.com/open-mmlab/mmengine/blob/main/docs/zh_cn/advanced_tutorials/visualization.md](https://github.com/open-mmlab/mmengine/blob/main/docs/zh_cn/advanced_tutorials/visualization.md) .
2. Familiarize yourself with the visualization implementation in MMagic, which can be found at [https://github.com/open-mmlab/mmagic/tree/main/mmagic/visualization](https://github.com/open-mmlab/mmagic/tree/main/mmagic/visualization) .
3. Implement tensorboard and wandb visualization in MMagic. Configure the visualization backend in the config file and successfully test it to ensure that it can save images, losses, etc. You can choose a model you are familiar with for testing, such as basicvsr, edsr, inpainting, matting, biggan, etc.
4. Write user guides in both English and Chinese. Create a documentation file at [https://github.com/open-mmlab/mmagic/blob/main/docs/en/user_guides/visualization.md](https://github.com/open-mmlab/mmagic/blob/main/docs/en/user_guides/visualization.md) .
5. Ensure that the user guide includes step-by-step instructions, code examples, and explanations of how to utilize MMagic's tensorboard and wandb visualization features effectively.

MMagic 目前基本支持了本地可视化（LocalVisualization），但是没有测试过MMEngine中的TensorboardVisualization和WandbVisualization。MMagic集成了MMGeneration的可视化后端，支持了视频可视化，但不保证所有模型可用。
1. 学习MMEngine可视化教程 [https://github.com/open-mmlab/mmengine/blob/main/docs/zh_cn/advanced_tutorials/visualization.md](https://github.com/open-mmlab/mmengine/blob/main/docs/zh_cn/advanced_tutorials/visualization.md) 
2. 熟悉MMagic可视化相关实现：[https://github.com/open-mmlab/mmagic/tree/main/mmagic/visualization](https://github.com/open-mmlab/mmagic/tree/main/mmagic/visualization) 
3. 实现MMagic的tensorborad和wandb可视化（指在config中配置可视化后端并成功测试，能保存图片、loss等），（可选取你熟悉的模型进行测试，推荐测试：basicvsr，edsr，inpaiting模型，matting模型，biggan等）
4.编写中英文使用教程 [https://github.com/open-mmlab/mmagic/blob/main/docs/en/user_guides/visualization.md](https://github.com/open-mmlab/mmagic/blob/main/docs/en/user_guides/visualization.md) 

## Task Information

**Repository**：MMagic

**Difficulty**：Medium

**Points**：30

**Estimated Time Consumption**：14 days

## Task Status

**Claimant's Github ID**：aptsunny

**Claim Time**：2023年07月11日

**PR Link**：https://github.com/open-mmlab/mmagic/pull/1939

## Points to Note

❗️请注意：此任务属于超级视客营任务，不能被领取！请登陆 [超级视客营平台](https://openmmlab.com/activity/codecamp) 按照流程申请任务，我们会尽快审核～

**Comment by aptsunny** [comment]
Claim the mission: https://github.com/open-mmlab/mmagic/pull/1939

**Reply by crazysteeaam** [reply]
good job~
